### PR TITLE
added FPS configuration options for RTSP, camera default is 15fps

### PR DESCRIPTION
**Pull request by claudobahn**
@gtxaspec I was looking to lower the output frame rate of at least the LOW_RES stream for use as a detection stream in Frigate.

I tested that the patch does indeed change the frame rate, but haven't tested the camera's behavior much, yet. With lower frame rates there's a lot more lag in the video.

### CODE REVIEW

**Comment by gtxaspec** [comment]
you're saying when you lower the FPS, the lag in the video stream increases?

**Comment by claudobahn** [comment]
@gtxaspec yes, when I have it set to 5fps (Frigate's preferred setting for the object detection stream) it does seem to lag 3-5 seconds behind. If I used `impdbg` to override the FPS to 15 for the low res stream, it temporarily catches back up, but then drops back down to 5fps (not sure what resets it back down) and resumes the increase in latency.

**Comment by claudobahn** [comment]
@gtxaspec speaking of `impdbg`, I noticed that the settings for target / max bit rates in `run_mmc.ssh` end up showing flipped in the output of `impdb --enc_info`.

According to the output of `impdb --enc_info`:
```
chnAttr->rcAttr->CappedQuality->uTargetBitRate = 128(0x80) offset:size = 48:4
chnAttr->rcAttr->CappedQuality->uMaxBitRate = 512(0x200) offset:size = 52:4
```

Target is register 48, Max is register 52.

But the `run_mmc.sh` uses them flipped:
```
	if [[ "$RTSP_HI_RES_MAX_BITRATE" != "" ]]; then
	watch -n10 -t "/system/bin/impdbg --enc_rc_s 0:48:4:$RTSP_HI_RES_MAX_BITRATE" > /dev/null 2>&1 &
	fi

	if [[ "$RTSP_HI_RES_TARGET_BITRATE" != "" ]]; then
	watch -n10 -t "/system/bin/impdbg --enc_rc_s 0:52:4:$RTSP_HI_RES_TARGET_BITRATE" > /dev/null 2>&1 &
	fi
```

I was looking at those more closely because I was trying to play with different options to see if I could reduce the latency by using a different encoding mode or bitrate.

**Comment by gtxaspec** [comment]
oops good catch, I'll get it fixed in the next release

**Comment by gtxaspec** [comment]
try changing the RC mode to 2, vbr and see if the lag gets better

**Comment by claudobahn** [comment]
I had tried (and did again) 2,4,8 but that didn't make a difference. Changing the bitrate also didn't really make a difference.

While playing with this, I did notice that changing the bitrate didn't really seem to affect the actual stream bitrate much. But what had an immediate impact was changing `iMinQP` (`58:2`). When I increased that to `40`, the picture got a lot worse but the bandwidth dropped to 1KB/s.

I don't have enough experience in this, but my suspicion is that the distribution of I/P/B frames could possibly play into the lag/latency problem.

All that said, I think this PR is functional as-is. And the 3-5 second latency on the stream is certainly no issue for me for an object detection stream.

**Comment by claudobahn** [comment]
@gtxaspec

Running the low res stream (in a high/low res setup) through:
`ffprobe -v info -show_frames rtsp://<camera_ip_pw>:8554/video2_unicast | grep pict_type`

shows it's actually only producing I and P frames, no B frames which I had originally suspected would introduce extra latency. The Group Of Pictures setting is 10 by default (1x I, followed by 9x P) and I tried to reduce that to 5, but that made no difference.

Basically what I'm observing is that the camera is queuing up about 20 frames before it will start the stream, regardless of FPS or GOP length setting. So the lower the FPS setting, the more latency this introduces and at 5fps that's about 4s latency before the stream starts.

**Comment by gtxaspec** [comment]
@claudobahn interesting, have you tried analyzing the .mp4 files to see if they are also missing B frames? just to make sure it's the camera, not the rtsp server

**Comment by claudobahn** [comment]
@gtxaspec I have not, will do.

Speaking of RTSP server... I should have looked at the settings for that. The queuing of frames is probably the RTSP server's doing and not the camera's. With the default GOP length of 10, it queues up 2, so 20 frames. But streams them all at once. With a shorter GOP length of 5, it seemed to queue up 5 groups, so 25 frames.

**Comment by claudobahn** [comment]
@gtxaspec I recorded a short clip using the Wyze app and also only got I and P frames.

fyi, looks like the GOP length defaults to 2x FPS. So at the default 15fps, the GOP length is actually 30 (1 "I" frame followed by "P" frames).

I then tried rebooting the camera w/ a reconfigured `v4l2rtspserver` -- I added `-Q 1` to shorten the frame queue. That seems to have done the trick.

**Comment by claudobahn** [comment]
@gtxaspec noticed that you've integrated my changes already. I've rebased this PR to only include the extra configuration statements in the README.

You should be good to merge this, I think?

**Comment by gtxaspec** [comment]
thanks!

**Comment by gtxaspec** [comment]
> @gtxaspec I recorded a short clip using the Wyze app and also only got I and P frames.
> 
> fyi, looks like the GOP length defaults to 2x FPS. So at the default 15fps, the GOP length is actually 30 (1 "I" frame followed by "P" frames).
> 
> I then tried rebooting the camera w/ a reconfigured `v4l2rtspserver` -- I added `-Q 1` to shorten the frame queue. That seems to have done the trick.

would this benefit @1080 also?

**Comment by claudobahn** [comment]
@gtxaspec are you asking if it would benefit the hires config, as well? I think so.

**Comment by gtxaspec** [comment]
@claudobahn do you think it would be worthwhile to overall performance to add -Q 1 to the default config (at full fps)?  ( i'm not a video guy yet...lol )

**Comment by claudobahn** [comment]
@gtxaspec still testing, I'll let you know what I find. I'll open a new PR for that. Video guy or not, your project is very useful :)

**Comment by gtxaspec** [comment]
also, these docs provide some insight on video stuff just in case you haven't seen them.  you can Google doc translate the pdfs too

https://github.com/BiatuAutMiahn/IVSP-T31-1.1.1-20200508/tree/main/software/doc
